### DISCUSSION TITLE
Building CBL-Mariner on Ubuntu docker container

### DISCUSSION THREAD

**Discussion by sunilchintu2468** [microsoft/azurelinux#4816] [Q&A]
Hi, I'm trying to build Mariner 2.0-stable on an Ubuntu docker container (started by root user with --privileged mode) by following the steps mentioned [here](https://github.com/microsoft/CBL-Mariner/blob/2.0/toolkit/docs/quick_start/quickstart.md#iso-image).

The build failed with the following error:
![image](https://user-images.githubusercontent.com/652389/217917095-141106fb-1335-43c9-b686-93b035a459fb.png)

Then I tried to set up a chroot directory and export the path to the CHROOT_DIR env variable:
![image](https://user-images.githubusercontent.com/652389/217917430-1429a2bd-8037-48e7-8526-a07e9a7f30ba.png)

The build process proceeded and failed after an hour with some more errors:
![image](https://user-images.githubusercontent.com/652389/217917694-39702786-b4c6-4cb2-8d19-9dbd99e49f48.png)

To avoid this, I created a lock file manually, not sure whether it has any other impacts:
![image](https://user-images.githubusercontent.com/652389/217917832-d78afb9d-bb5e-4cc0-89d4-c29ca5f43eaf.png)

Build proceeded further and failed again :
![image](https://user-images.githubusercontent.com/652389/217918168-abc1962d-e4fe-43d8-b12f-2da3fb65806b.png)

Not understanding the root cause of these failures, can anybody please help?

**Comment by nicogbg** [comment]
I'm not sure building image from a container is supported or easy. CBL-Mariner toolkit uses raw disk mechanism to build images which is, as far as I can remember, is not allowed in "regular" container. Making the container privileged might help though.

**Comment by mfrw** [comment]
More details with respect to a similar discussion:
- https://github.com/microsoft/CBL-Mariner/issues/6023
